src/tools/fuzzing.h

```diff
@@ -780,11 +777,12 @@ class TranslateToFuzzReader {
     std::vector<Expression*> invocations;
     while (oneIn(2) && !finishedInput) {
       std::vector<Expression*> args;
-      for (auto type : func->params) {
+      for (auto type : func->sig.params.expand()) {
```

Comment:
Ah, this is a good idea. Part of the reason I was ok writing `expand()` everywhere is because it is relatively expensive (it acquires a rw lock for reading) so it is good to call it as few times as possible. I think we may still want to expose `expand()` instead of implementing `[]` to avoid hiding that cost, but I think it would be possible to implement the iterators efficiently. I could also be convinced to implement `[]` as well.

Do you mind if we leave this for a follow up PR, though?

Reply:
I'm not sure exposing the function will make people use it less... Unless we are only checking the size of a type, if we need the info of independent single type element of a type, we should do that anyway whether or not `expand` is exposed, no?

Anyway I was not suggesting we do this in this PR :)